core/src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -329,6 +350,12 @@ case class MergeIntoCommand(
             isOverwriteMode = false, rearrangeOnly = false)
         }
 
+        // If materialized, prepare the DF reading the materialize source
+        // Otherwise, prepare a regular DF from source plan.
+        val materializeSourceReason =
+        createSourcePlanDF(spark, source, condition, matchedClauses, notMatchedClauses,
```

Comment:
it would be good to name this method such that its obvious that it returns a reason. `createX` named method typically imply that it returns `X`.